apis/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -228,6 +266,40 @@ type USMFeatureConfig struct {
 	Enabled *bool `json:"enabled,omitempty"`
 }
 
+// DogstatsdConfig contains the Dogstatsd configuration parameters.
+// +k8s:openapi-gen=true
+type DogstatsdConfig struct {
+	// Enable origin detection for container tagging.
+	// See also: https://docs.datadoghq.com/developers/dogstatsd/unix_socket/#using-origin-detection-for-container-tagging
+	// +optional
+	OriginDetection *bool `json:"originDetection,omitempty"`
+
+	// HostPortConfig contains host port configuration.
+	// Enabled Default: true
+	// Port Default: 8125
+	// +optional
+	HostPortConfig *HostPortConfig `json:"hostPortConfig,omitempty"`
+
+	// UnixDomainSocketConfig contains socket configuration.
+	// See also: https://docs.datadoghq.com/agent/kubernetes/apm/?tab=helm#agent-environment-variables
+	// Enabled Default: true
+	// Path Default: `/var/run/datadog/apm.socket`
+	// +optional
+	UnixDomainSocketConfig *UnixDomainSocketConfig `json:"unixDomainSocketConfig,omitempty"`
+
+	// Configure the Dogstasd Mapper Profiles.
+	// Can be passed as raw data or via a json encoded string in a config map.
+	// See also: https://docs.datadoghq.com/developers/dogstatsd/dogstatsd_mapper/
+	// +optional
+	MapperProfiles *CustomConfig `json:"mapperProfiles,omitempty"`
+}
+
+// EventCollectionConfig contains the Event Collection configuration.
+// +k8s:openapi-gen=true
+type EventCollectionConfig struct {
+	KubernetesEventsCollection *bool `json:"kubernetesEventsCollection,omitempty"`
```

Comment:
```suggestion
	CollectKubernetesEvents *bool `json:"collectKubernetesEvents,omitempty"`
```
suggesting an "active" format to be consistent with other `bool`s. 
I assume the idea is to support additional event types in the future?